cpp/CMakeLists.txt

```diff
@@ -149,20 +152,27 @@ if (enable-test)
   include_directories(${GTEST_INCLUDE_DIRS})
 
   # Add source files according to machine setup
-  if (CUDA_HAVE_GPU)
+  if (CUDA_AND_GPU)
+  message("MAKING GPU TESTS")
     file(GLOB_RECURSE TEST_SRC_FILES RELATIVE
          ${PROJECT_SOURCE_DIR}
          test/cpu_operations_test/*.cc
-         test/util_test/*.cc
+         test/logistic_regression_test/*.cc
+         test/gpu_logistic_regression_test/*.cc
+	       test/util_test/*.cc
          test/cpu_solver_test/*.cc
          test/gpu_operations_test/*.cc
          test/gpu_solver_test/*.cc
-         test/model_test/*.cc
+         test/matrix_vector_mult_test/*.cc
```

Comment:
And move matrix vector multiplication related tests to NICE/cpp/test/gpu_operations_test

Reply:
Fixed.